### ISSUE TITLE
Improve the naming of Host and IP configuration settings

### ISSUE THREAD

**Issue by hayley-jean** [EventStore/EventStore#3767]
**Is your feature request related to a problem? Please describe.**
There are a number of options for configuring the host and ip of a node, what is advertised in the gossip, and what addresses are used to gossip with other nodes. Misconfiguration of these settings results in the cluster not working, or in duplicate node entries in the gossip.

The option names do not make it clear what they will affect, and they do not guide the user in correctly configuring the cluster.

**Describe the solution you'd like**
Rename the options, or improve the way these interfaces are configured so that it is easier to find out what to do (and where the configuration may be wrong in the case of failures).

Previously, cluster communication (gossip, elections, and replication) used to only run over `IntIp`, and external connections used `ExtIp`. But since 20.10, only TCP replication happens over the `IntIp` and all other communication happens over `ExtIp`.
We should update the configuration to make this new distinction clear.

As a suggestion, we could rename the `Int` options to `Replication`. For example:

| Current | Proposed |
|---------|------------|
| IntIp | ReplicationIp |
| IntHostAdvertiseAs | ReplicationHostAdvertiseAs |
| IntTcpHeartbeatInterval | ReplicationHeartbeatInterval |

And rename the `Ext` options to `Node`. For example:

| Current | Proposed |
|----------|-----------|
| ExtIp | NodeIp |
| ExtHostAdvertiseAs | NodeHostAdvertiseAs |
| HttpPortAdvertiseAs | NodePortAdvertiseAs |

**Additional context**
We need to retain rolling upgrade support, so instead of renaming or removing the existing options, we would need to deprecate them first.

It may also be worth looking into whether there's a clearer way to name the `AdvertiseAs` options.

**Comment by ylorph** [comment]
Will there be a period of time where both names will be supported ?

( thinking of people having automation in place )

**Comment by alexeyzimarev** [comment]
It would be nice to add those settings to a patch version as a non-breaking change, maybe.

**Comment by alexeyzimarev** [comment]
We were discussing those settings, and when I tried to explain the difference between `ExtHostAdvertiseAs` and `AdvertiseToClientsAs`, I wasn't able to do so. I remember we had a good reason for it, but can't remember what it was. @hayley.campbell do you know when those two would be different?

**Comment by alexeyzimarev** [comment]
Yeah, that's more or less what I thought, but I don't understand the meaning of `ExtHostAdvertiseAs` being used for internal gossip when `IntHostAdvertiseAs` is configured. It seems like when `IntHostAdvertiseAs` *and* `AdvertiseToClientsAs` are configured, `ExtHostAdvertiseAs` becomes meaningless. Is that what we want to solve by the new names? With `NodeHostAdvertiseAs`, would `AdvertiseToClientsAs` be still needed?
